consensus/polybft/checkpoint_manager.go

```diff
@@ -170,13 +205,13 @@ func (c *checkpointManager) submitCheckpoint(latestHeader types.Header, isEndOfE
 		}
 	}
 
-	return c.encodeAndSendCheckpoint(txn, latestHeader, *currentExtra, isEndOfEpoch)
+	return c.encodeAndSendCheckpoint(txn, latestHeader, currentExtra, isEndOfEpoch)
 }
 
 // encodeAndSendCheckpoint encodes checkpoint data for the given block and
 // sends a transaction to the CheckpointManager rootchain contract
 func (c *checkpointManager) encodeAndSendCheckpoint(txn *ethgo.Transaction,
-	header types.Header, extra Extra, isEndOfEpoch bool) error {
```

Comment:
Not against this change. However, the reason I didn't pass pointers but values is because I wanted to explicitly suggest that those header parameters shouldn't be mutated in this function (but rather read-only parameters).

Reply:
But that is also what we are trying to do with these `components` (`handlers`) in RFC-176. You are getting immutable data, so you can create a snapshot of your state. I don't think there is a need to change a state of any object passed to these components, nor it should be.